src/qibolab/platforms/abstract.py

```diff
@@ -175,58 +175,88 @@ def transpile(self, circuit: Circuit):
+                    sequence.virtual_z_phases[qubit] += np.pi
+
+                elif isinstance(gate, gates.RZ):
+                    qubit = gate.target_qubits[0]
+                    sequence.virtual_z_phases[qubit] += gate.parameters[0]
+
+                elif isinstance(gate, gates.U3):
+                    qubit = gate.target_qubits[0]
+                    # Transform gate to U3 and add pi/2-pulses
+                    theta, phi, lam = gate.parameters
+                    # apply RZ(lam)
+                    sequence.virtual_z_phases[qubit] += lam
+                    # Fetch pi/2 pulse from calibration
+                    RX90_pulse_1 = self.create_RX90_pulse(
+                        qubit,
+                        start=max(sequence.get_qubit_pulses(qubit).finish, moment_start),
+                        relative_phase=sequence.virtual_z_phases[qubit],
+                    )
+                    # apply RX(pi/2)
+                    sequence.add(RX90_pulse_1)
+                    # apply RZ(theta)
+                    sequence.virtual_z_phases[qubit] += theta
+                    # Fetch pi/2 pulse from calibration
+                    RX90_pulse_2 = self.create_RX90_pulse(
+                        qubit, start=RX90_pulse_1.finish, relative_phase=sequence.virtual_z_phases[qubit] - np.pi
+                    )
+                    # apply RX(-pi/2)
+                    sequence.add(RX90_pulse_2)
+                    # apply RZ(phi)
+                    sequence.virtual_z_phases[qubit] += phi
+
+                elif isinstance(gate, gates.M):
+                    # Add measurement pulse
+                    measurement_start = max(sequence.get_qubit_pulses(*gate.target_qubits).finish, moment_start)
+                    gate.pulses = ()
+                    for qubit in gate.target_qubits:
+                        MZ_pulse = self.create_MZ_pulse(qubit, start=measurement_start)
+                        sequence.add(MZ_pulse)
+                        gate.pulses = (*gate.pulses, MZ_pulse.serial)
+
+                elif isinstance(gate, gates.CZ):
+                    # create CZ pulse sequence with start time = 0
+                    cz_sequence = self.create_CZ_pulse_sequence(gate.qubits)
+
+                    # determine the right start time based on the availability of the qubits involved
+                    cz_qubits = {*cz_sequence.qubits, *gate.qubits}
+                    cz_start = max(sequence.get_qubit_pulses(*cz_qubits).finish, moment_start)
+
+                    # shift the pulses
+                    for pulse in cz_sequence.pulses:
+                        if pulse.se_start.is_constant and pulse.se_duration.is_constant:
+                            pulse.start += cz_start
+                        else:
+                            raise NotImplementedError(
+                                f"Shifting start times of pulses using symbolic expressions is not supported yet"
+                            )
```

Comment:
This error is added in the scenario where we use `symbolic` in the runcard instead of the relative starts?

Reply:
Thanks @maxhant,
I agree, this isn't really necessary, as the pulses created in `create_CZ_pulse_sequence()` will not use symbolic expressions. I will remove it.